third_party/tlx/tutorials/gemm-WS-hopper.py

```diff
@@ -0,0 +1,247 @@
+import pytest
+import torch
+
+import triton
+import triton.language as tl
+import triton.tlx.language as tlx
+from typing import Optional
+from triton.tools.tensor_descriptor import TensorDescriptor
+
+DEVICE = triton.runtime.driver.active.get_active_torch_device()
+
+
+def is_cuda():
+    return triton.runtime.driver.active.get_current_target().backend == "cuda"
+def is_hip_cdna2():
+    target = triton.runtime.driver.active.get_current_target()
+    return target.backend == 'hip' and target.arch == 'gfx90a'
+
+def get_cuda_autotune_config():
+    return [
+        triton.Config({'BM': 128, 'BK': 128, 'BN': 128},
```

Comment:
NUM_STAGES should be tunable too. Also is the number of consumers. Please see https://github.com/facebookexperimental/triton/pull/205.